src/huggingface_hub/hf_file_system.py

```diff
@@ -368,6 +373,11 @@ def _ls_tree(
                 out.append(cache_path_info)
         return out
 
+    def glob(self, path, maxdepth=None, **kwargs):
+        # Set expand_info=False by default to get a x10 speed boost
+        kwargs = {"expand_info": False, **kwargs}
```

Comment:
```suggestion
        kwargs = {"expand_info": detail or False, **kwargs}
```
This value should be `True` if `detail=True` to be consistent with the spec. 

We need more tricks to optimize the glob logic in `datasets`  anyway.

Reply:
Good catch ! I fixed it with https://github.com/huggingface/huggingface_hub/pull/1815/commits/1cfb2f19cb2d9f64f21e4ed0e7dd7c8b4fa8519e